snap/backend/src/utils/accountUtils.js

```diff
@@ -0,0 +1,30 @@
+import { getBIP44AddressKeyDeriver } from '@metamask/key-tree';
+import { PrivateKey } from 'symbol-sdk';
+import { Network, SymbolFacade } from 'symbol-sdk/symbol';
+
+const accountUtils = {
+	/**
+	 * * Derives a key pair from a mnemonic and an address index.
+	 * @param {104 | 152} identifier - The network identifier.
+	 * @param {number} addressIndex - The address index.
+	 * @returns {Promise<SymbolFacade.KeyPair>} - The derived key pair.
+	 */
+	async deriveKeyPair(identifier, addressIndex) {
+		const coinType = Network.MAINNET.identifier === identifier ? 4343 : 1;
```

Comment:
coin type is defined in [SLIP-0044](https://github.com/satoshilabs/slips/blob/master/slip-0044.md)
`4343` represents Symbol
`1` Testnet (all coins)

>  Should these be in an enum/const like object?
sure, I will update it 👍🏼

Reply:
can you use this function instead?
https://github.com/symbol/symbol/blob/dev/sdk/javascript/src/facade/SymbolFacade.js#L364

Reply:
> can you use this function instead? https://github.com/symbol/symbol/blob/dev/sdk/javascript/src/facade/SymbolFacade.js#L364

sure, so to get the **coinType** it will be something like this `facade.bip32Path(0)[1]`